### PR TITLE
Add guide how to work with local registry

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR adds a new section in the docs which describes how to setup for AEA development.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: docs/development-setup.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## Development setup
+
+To prepare a directory (henceforth working directory) for development with the AEA framework you can take a few steps:
+
+- Ensure you start with an empty working directory to avoid any unnecessary side-effects.
+
+- In your working directory, create an empty folder called `packages`. This folder will act as the local registry for packages.
+
+- In your working directory, create a `.env` file with the constant `PYTHONPATH=path_to_packages_dir` where `path_to_packages_dir` is the path to the packages folder in your working directory.
```
Comment (marcofavorito) [inline docs/development-setup.md:13]: two comments:
1. `PYTHONPATH=path_to_packages_dir` will overwrite the old `PYTHONPATH`, that causes problems if the project environment relied on the content of `PYTHONPATH`
2. the path should point to the _parent directory_ of the `packages` directory

Comment (DavidMinarsch) [reply docs/development-setup.md:13]: Re 2 see further down. Re 1. we need to modify it, so what do you suggest then instead.

Comment (DavidMinarsch) [reply docs/development-setup.md:13]: Btw we tested this on @jrriehl 's machine and it worked fine

File: docs/development-setup.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+- Depending on your editor, you might take further steps:
+
+	- VS Code: The Python Extension in VS Code can be configured to include additional paths in the Python path. The extension has a setting for `python.envFile` which specifies the path to a file containing environment variable definitions. The default is set to `"python.envFile": "${workspaceFolder}/.env"`. Provide the path to the `.env` file in the above settings. In the `.env` file, add the `PYTHONPATH` constant defined above. Then close VS Code and re-open it for the settings to take effect.
+
+## Developing an AEA
+
+- When developing a specific AEA, it might be helpful to publish/push or fetch/add from local registry. From your working directory, simply execute the usual AEA CLI commands. The CLI will first search in the `packages` directory, then in the remote AEA registry. You can explicitly point to local registry by providing flag `--local` (see <a href="../cli-commands">here</a>)
```
Comment (marcofavorito) [inline docs/development-setup.md:21]: ... or `--remote` to only point to remote registry

File: docs/development-setup.md (hunk 1 above)
Comment (marcofavorito) [reply docs/development-setup.md:13]: we agreed to change the instruction with `PYTHONPATH=path_to_working_dir`.

Comment (marcofavorito) [reply docs/development-setup.md:13]: regarding 1: try whether `PYTHONPATH=$PYTHONPATH:path_to_working_dir` works. If yes, that should be preferable.

Comment (jrriehl) [reply docs/development-setup.md:13]: Yes, this works. I just checked.

Comment (jrriehl) [reply docs/development-setup.md:13]: Sorry, to be more specific, what worked was `PYTHONPATH=$PYTHONPATH:path_to_packages_dir`.

The above suggestion did not work.

**Review by jrriehl — APPROVED** [review]
LGTM

**Comment by jrriehl** [comment]
Looks good, I tried out the template repo and it seems to work for me.

File: docs/development-setup.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Developing an AEA
+
+- When developing a specific AEA, it might be helpful to publish/push or fetch/add from local registry. From your working directory, simply execute the usual AEA CLI commands. The CLI will first search in the `packages` directory, then in the remote AEA registry. You can explicitly point to local registry by providing flag `--local` (see <a href="../cli-commands">here</a>) or `--remote` to only point to remote registry.
+
+- When working on an AEA it can help to provide a symbolic link to the packages directory, so that the import paths are detected by your editor. Simply create an empty file with `touch packages` in your AEA project, then create a symbolic link to the `packages` directory with `ln -s ../packages packages`.
```
Comment (danielksan81) [inline docs/development-setup.md:29]: Here you mean inside of the aea folder and not the workspace folder? That also implies that inside of the packages folder needed dependencies are available. Using the _aea add_ command add dependencies inside of vendor folder by default, so this has to be added in addition from my point of view.

**Pull request merged**